### ISSUE TITLE
LightwaveRf Binding Conversino Error

### ISSUE THREAD

**Issue by oz1871** [foxy82/openhab#28]
Guys, 
My binding is working for all parts and I am getting a message every 20 seconds or so.
I am running latest stable binary I have the rules added for the Energy Monitor in my Items File but nothing is working?

Example Message:
[ERROR] [l.internal.LightwaveRfWifiLink] - Error converting message: *!{"trans":87985,"mac":"20:2B:2B","time":1453826435,"prod":"pwrMtr","serial":"BC2FFE","router":"000500","type":"energy","cUse":595,"todUse":9826,"yesUse":0}

Many Thanks, 

Owen 

Great work by the way/

**Comment by foxy82** [comment]
Hi,

I assume you've been upgraded to the latest lightwave firmware U2.91Z? 

Unfortunately the fixes for this have been made but are not merged into Openhab as there was no one able to test the binding on Openhab2. The pull request is here: https://github.com/openhab/openhab/pull/3457

I've created an ad hoc binary here that you can use but you'll need to watch the forums for any updates: https://community.openhab.org/t/lightwave-binding-broken-by-latest-firmware/5914

Regards,
Neil

**Comment by oz1871** [comment]
I have started using the Alpha 10 Release, Ive tried it on both OpenHab 1.8 and 2 and I am still getting the message exactly the same as the one posted above. Ill wait for some updates, if you need anyone to test let me know I have all of the lightwave products Im not using them all yet but I have access to them all.

Regards, 
Owen

**Comment by foxy82** [comment]
Hi, try the one here: https://github.com/foxy82/openhab/releases/tag/1.8.0_alpha11

**Comment by oz1871** [comment]
That release is working again with both releases again Neil. Ill test all the variables in a sitemap to confirm but looks good so far.

**Comment by oz1871** [comment]
Ive hooked up the Thermostat and boiler switch today and I am getting two new errors.
Thermostat:
2016-02-02 13:24:47.344 [ERROR] [l.internal.LightwaveRfWifiLink] - Error converting message: :202B2B,796,!R3DhF*r|Requesting|Temperature
2016-02-02 13:28:46.077 [WARN ] [.l.internal.LightwaveRfBinding] - No item for incoming message[798,!R3DhF*tP17.5]
2016-02-02 13:28:48.440 [ERROR] [l.internal.LightwaveRfWifiLink] - Error converting message: :202B2B,799,!R3DhF*r|Requesting|Temperature

Boiler Switch:
2016-02-02 13:26:41.846 [ERROR] [l.internal.LightwaveRfWifiLink] - Error converting message: :202B2B,797,!R2DhF*r|Requesting|Temperature

Anyway I can help you work out binding additions for these? The app looks like it can pull some useful information. I would imagine they are similar to the radiator valves.

**Comment by oz1871** [comment]
Chaging the Thermostat Mode:
2016-02-02 13:47:55.607 [ERROR] [l.internal.LightwaveRfWifiLink] - Message not recognised: :202B2B,805,!R3DhF*mP0|Standby|Mode set

**Comment by oz1871** [comment]
I have done lots of packet sniffing with wireshark there are no indications of a serial as it sends the command just the R3DhF* and R2DhF* I have these setup in the same room on my App.

**Comment by foxy82** [comment]
Hi,

I assume these are the messages sent from an iPhone or Android app when executing particular functions? Does the wifi link send any kind of reply message that is where the serial number is for the radiator valves?

Also what are all the modes for the thermostat?

Finally any idea why you have two different syntaxs for these two messages:
:202B2B,796,!R3DhFr|Requesting|Temperature
:202B2B,799,!R3DhF*r|Requesting|Temperature

Neil

**Comment by oz1871** [comment]
Hi Neil, 
This is from the iPhone App
There doesn't seem to be a response packet from the device. If I wireshark the IP and monitor the UDP traffic on ports 9760 and 9761 I only see traffic from 9761

Example Trace:
:202B2B,100,!F*p
:202B2B,452,!R5DhF*r|Requesting|Temperature
:202B2B,100,!F*p
:202B2B,100,!F*p
:202B2B,100,!F*p
:202B2B,453,!R5DhF*mP4|Constant|Mode set
:202B2B,100,!F*p
:202B2B,100,!F*p
:202B2B,454,!R5DhF*mP2|Away|Mode set
:202B2B,100,!F*p
:202B2B,100,!F*p
:202B2B,100,!F*p
:202B2B,100,!F*p
:202B2B,100,!F*p
:202B2B,100,!F*p

The Modes from the Top 
Standby
Running
Away
Frost
Constant
Holiday  
Seems to Range from mP0 to mP5

The difference in the *r and the none *r is when I change the temperature which sends a R3DhFtP followed by the value of the temperature see example above, it then sends a *r update message afterwards where as if I ask the thermostat or boiler switch for an update in the app manually its a *r I will thoroughly test this tomorrow. In the App it seems to be able to report on 
Thermostat in the App Shows: 
Current Temp X.XC
Target Temp X.XC
Next Change  XX:XX (24 hour time)
Next Target Temp X.XC
Output XXX (ON/OFF)
Battery (Battery Image)
Last Updated Time  Full Date.

Once you tap on it there are also lots of ways to set programmable Days and Patterns and Temp Levels.
Ill see if these output anything in the debug logs.

The Boiler Switch in App shows:
Current Temp X.XC
Target Temp X.XC
Next Change XX:XX(24 hour time)
Next Target (ON/OFF)
Output XXX (ON/OFF)

When  you go into this one its very similar but it allows you to set an ON / OFF pattern or you can switch it to Temp level so this can be constant on/off or be controlled via temperature patterns.

They both have an ambient Temp Sensor built in as it will display temp of the room when in Rooms View.

**Comment by oz1871** [comment]
20:28:56.704 [WARN ] [.l.internal.LightwaveRfBinding:358  ] - No item for incoming message[*!{"trans":354,"mac":"20:2B:2B","time":1456432136,"prod":"tmr1ch","serial":"BD6902","router":"202B2B","type":"temp","batt":2.49,"ver":33,"state":"run","cTemp":19.3,"cTarg":18.0,"output":0,"nTarg":17.0,"nSlot":"22:00","prof":4}]

I have looked deeper into the Binding and found some more useful info for you this is the output from the thermostat.

cTarg is Current Temperature Target nTarg is Next Temperature Target and nSlot is Next Time slot to update.

**Comment by nxfifteen** [comment]
I'm also having problems with conversions from my energy meter - see error logs bellow - I've tried with both the stable **org.openhab.binding.lightwaverf-1.8.3.jar** version and **org.openhab.binding.lightwaverf-1.8.0-SNAPSHOT.jar ALPHA12** with no success.

Do you have any solutions? I see from the previous comments you still waiting on openhab#3457 - I dont have the ability/skill to build the source, do you have a compiled newer version than **ALPHA12**

2016-05-24 17:32:28.567 [ERROR] [l.internal.LightwaveRfWifiLink] - Error converting message: *!{"trans":1257,"mac":"00:00:00","time":1464111147,"prod":"pwrMtr","serial":"AAAAAA","router":"000500","type":"energy","cUse":839,"todUse":4941,"yesUse":0}
2016-05-24 17:32:43.562 [ERROR] [l.internal.LightwaveRfWifiLink] - Error converting message: *!{"trans":1258,"mac":"00:00:00","time":1464111162,"prod":"pwrMtr","serial":"AAAAAA","router":"000500","type":"energy","cUse":662,"todUse":4944,"yesUse":0}
2016-05-24 17:32:58.558 [ERROR] [l.internal.LightwaveRfWifiLink] - Error converting message: *!{"trans":1259,"mac":"00:00:00","time":1464111177,"prod":"pwrMtr","serial":"AAAAAA","router":"000500","type":"energy","cUse":644,"todUse":4946,"yesUse":0}
2016-05-24 17:33:13.553 [ERROR] [l.internal.LightwaveRfWifiLink] - Error converting message: *!{"trans":1260,"mac":"00:00:00","time":1464111192,"prod":"pwrMtr","serial":"AAAAAA","router":"000500","type":"energy","cUse":634,"todUse":4949,"yesUse":0}
2016-05-24 17:33:28.549 [ERROR] [l.internal.LightwaveRfWifiLink] - Error converting message: *!{"trans":1261,"mac":"00:00:00","time":1464111207,"prod":"pwrMtr","serial":"AAAAAA","router":"000500","type":"energy","cUse":648,"todUse":4952,"yesUse":0}
2016-05-24 17:33:43.544 [ERROR] [l.internal.LightwaveRfWifiLink] - Error converting message: *!{"trans":1262,"mac":"00:00:00","time":1464111222,"prod":"pwrMtr","serial":"AAAAAA","router":"000500","type":"energy","cUse":627,"todUse":4954,"yesUse":0}
